pkg/k8sutil/es_crud.go

```diff
@@ -0,0 +1,127 @@
+
+
+package k8sutil
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+	"github.com/Sirupsen/logrus"
+)
+
+
+/* 
+TODO:
+  Assuming orginal settings are default values like delayed_timeout is 1m and translog.durability is async, incase if the user is changed then 
+  user changed settings will be lost after scaling operations.
+  this can be corrected by reading the setting before change.
+*/
+
+func es_change_settings(es_ip , duration,translog_durability string)(error){  
+	var ret error
+	ret = errors.New("Scaling: error in setting delay timeout")
+	
+	logrus.Infof("Scaling: change ES setting:  delay timeout:%s  and translog durability: %s", duration, translog_durability)
+	client := &http.Client{
+	}
+	body:="{ "
+	body = body + "\"settings\": { \"index.unassigned.node_left.delayed_timeout\": \""+duration+"\" ,"
+	body = body + "\"index.translog.durability\": \""+translog_durability +"\" }"
+	body = body +  " }"
+
+	req, _ := http.NewRequest("PUT", "http://"+es_ip+"/_all/_settings", bytes.NewBufferString(body))
+	req.Header.Add("Content-Type", `application/json`)
+	resp, _ := client.Do(req)
+	data, _ := ioutil.ReadAll(resp.Body)
+
+	if (resp.StatusCode == 200){
+		ret = nil
+	}else{
+		if (strings.Contains(string(data), "index_not_found_exception")){
+			ret = nil  /* if there are no indexes , then this function need po pass */
+			fmt.Println("WARNING in setting delaytimeout: response: ",resp, string(data))
+			return ret
+		}
+		//string str:= "Scaling: setting delaytimeout: response: " + resp + string(data)
+		ret = errors.New("Scaling: setting delaytimeout: response: ")
+	}
+	return ret;
+}
+func es_checkForGreen(es_ip string)(error){ /* TODO */
```

Comment:
My original thoughts around the ES status was to make a set of stats that are broken out. Doing this allows us to use the same ES status checks and also enable other features like rolling restarts.

Reply:
Can you pls provide more details on this, Do you expect any change  here.